src/frontend/Navigation/Stack/index.tsx

```diff
@@ -70,6 +72,16 @@ function RootStackNavigatorChild() {
     },
   });
 
+  const security = useSecurityContext();
+
+  const {navigate} = useNavigationFromRoot();
+
+  React.useEffect(() => {
```

Comment:
nit: convert to `useLayoutEffect` so this happens slightly sooner in the react life cycle

Reply:
I think using `useLayoutEffect` causes issues here because what's happening inside this unfortunately needs the navigator to mount first. My understanding is that `useLayoutEffect` will attempt to run before that happens, which unfortunately doesn't play well with some of the navigation complexities this relies on.